server/service/src/processors/transfer/requisition/create_response_requisition.rs

```diff
@@ -0,0 +1,160 @@
+            &new_response_requisition.id,
+            &request_requisition.requisition_row,
+        )?;
+
+        RequisitionRowRepository::new(connection).upsert_one(&new_response_requisition)?;
+
+        let requisition_line_row_repository = RequisitionLineRowRepository::new(connection);
+
+        for line in new_requisition_lines.iter() {
+            requisition_line_row_repository.upsert_one(line)?;
+        }
+
+        let result = format!(
+            "requisition ({}) lines ({:?}) source requisition ({})",
+            new_response_requisition.id,
+            new_requisition_lines.into_iter().map(|r| r.id),
+            request_requisition.requisition_row.id
+        );
+
+        Ok(Some(result))
+    }
+}
+
+fn generate_response_requisition(
+    connection: &StorageConnection,
+    request_requisition: &Requisition,
+    record_for_processing: &RequisitionTransferProcessorRecord,
+) -> Result<RequisitionRow, RepositoryError> {
+    let store_id = record_for_processing.other_party_store_id.clone();
+    let name_id = request_requisition.store_row.name_id.clone();
+
+    let request_requisition_row = &request_requisition.requisition_row;
+
+    let requisition_number =
+        next_number(connection, &NumberRowType::ResponseRequisition, &store_id)?;
+
+    let result = RequisitionRow {
+        id: uuid(),
+        requisition_number,
+        name_id,
+        store_id,
+        r#type: RequisitionRowType::Response,
+        status: RequisitionRowStatus::New,
+        created_datetime: Utc::now().naive_utc(),
+        their_reference: request_requisition_row.their_reference.clone(),
+        max_months_of_stock: request_requisition_row.max_months_of_stock.clone(),
+        min_months_of_stock: request_requisition_row.min_months_of_stock.clone(),
+        // 5.
+        linked_requisition_id: Some(request_requisition_row.id.clone()),
+        expected_delivery_date: request_requisition_row.expected_delivery_date,
+        // Default
+        user_id: None,
+        sent_datetime: None,
+        finalised_datetime: None,
+        colour: None,
+        comment: None,
+    };
+
+    Ok(result)
+}
+
+fn generate_response_requisition_lines(
+    connection: &StorageConnection,
+    response_requisition_id: &str,
+    request_requisition: &RequisitionRow,
+) -> Result<Vec<RequisitionLineRow>, RepositoryError> {
+    let request_lines = get_lines_for_requisition(connection, &request_requisition.id)?;
+
+    let response_lines = request_lines
+        .into_iter()
+        .map(|l| l.requisition_line_row)
+        .map(
+            |RequisitionLineRow {
+                 id: _,
+                 requisition_id: _,
+                 item_id,
+                 requested_quantity,
+                 suggested_quantity,
+                 supply_quantity: _,
+                 available_stock_on_hand,
+                 average_monthly_consumption,
+                 snapshot_datetime,
+                 // TODO should comment transfer ?
```

Comment:
wondered the same - and as usual, I ask WWMD? (what would mSupply do?)

```
	[requisition]comment:=Get indexed string(17001; 175)+$t_serial+Choose([requisition]comment=""; ""; " ("+[requisition]comment+")")  // "From request requisition N (comment - if any)"
	[requisition]requester_reference:=Get indexed string(17001; 175)+$t_serial+Choose([requisition]requester_reference=""; ""; " ("+[requisition]requester_reference+")")  // "From request requisition N (requester_reference - if any)"
```

Reply:
🙏 

But I feel sooo ify about this, I've made a action plan under `Questions`[in this issue](https://github.com/openmsupply/open-msupply/issues/608)